src/Fixer/Operator/NoUselessConcatOperatorFixer.php

```diff
@@ -336,4 +348,27 @@ private function containsLinebreak(Tokens $tokens, int $startIndex, int $endInde
 
         return false;
     }
+
+    /**
+     * @param array{
+     *     start: int,
+     *     end: int,
+     *     type: self::STR_*,
+     * } $firstOperand
+     * @param array{
+     *     start: int,
+     *     end: int,
+     *     type: self::STR_*,
+     * } $secondOperand
+     */
+    private function operandsCanNotBeMerged(Tokens $tokens, array $firstOperand, array $secondOperand): bool
+    {
+        // If the first operand ends with a variable and the second operand does not start with a space,
+        // the variable in the first operand may be broken by the concatenation.
+
+        // example: "$param" . "hello" -> "$paramhello"
+
+        return $tokens[$firstOperand['end'] - 1]->isGivenKind(T_VARIABLE)
+            && !str_starts_with($tokens[$secondOperand['start'] + 1]->getContent(), ' ');
```

Comment:
this is too strict - the next token content must not start with identifier char

Reply:
Added tests bdd1ebc

Reply:
... And the logic has been changed a lot. 

It may be a little tricky, but it checks for unknown variables appearing after the operands are merged.

Reply:
...But again, the original strict logic may be more certain. 

For example, I can list the following other codes that should not be fixed, and it is difficult for me to say for sure that there are not more than these. Hmmm.

```php
echo "abc $d" . "[$e]";
echo "abc $d" . "[3]";
echo "abc $d" . '[3]';
echo "abc $d" . "->e";
echo "abc $d" . '->e';
echo "abc $d" . "->$e";
echo "abc $d" . "?->e";
echo "abc $d" . "?->$e";
echo "abc $d" . '?->e';
```

Reply:
3b0e1a6 Changed to a whitelist approach so that the fixer can fix safety and certainly

28edf77 And as pointed out, added an integration test after adjusting the priority of the existing fixer